client/gutenberg/extensions/markdown/editor.js

```diff
@@ -12,47 +12,48 @@ import { registerBlockType } from '@wordpress/blocks';
  * Internal dependencies
  */
 import './editor.scss';
-import edit from './edit';
-import save from './save';
-
-registerBlockType( 'a8c/markdown', {
-	title: __( 'Markdown' ),
-
-	description: (
-		<Fragment>
-			<p>{ __( 'Write your content in plain-text Markdown syntax.' ) }</p>
-			<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
-				{ __( 'Support reference' ) }
-			</ExternalLink>
-		</Fragment>
-	),
-
-	icon: (
-		<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
-			<rect
-				width="198"
-				height="118"
-				x="5"
-				y="5"
-				ry="10"
-				stroke="#000"
-				strokeWidth="10"
-				fill="none"
-			/>
-			<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
-		</svg>
-	),
-
-	category: 'jetpack',
-
-	keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
-
-	attributes: {
-		//The Markdown source is saved in the block content comments delimiter
-		source: { type: 'string' },
-	},
-
-	edit,
-
-	save,
-} );
+
+Promise.all( [ import( './edit' ), import( './save' ) ] ).then(
+	( [ { default: edit }, { default: save } ] ) =>
+		registerBlockType( 'a8c/markdown', {
```

Comment:
Note that if this loads after loading the post then the blocks may appear broken.

Can we verify on a slow connection?

Reply:
That's a good point. If the block is late to register, it's transformed to classic.

<details>
<summary>diff to test late load</summary>

```diff
diff --git a/client/gutenberg/extensions/markdown/editor.js b/client/gutenberg/extensions/markdown/editor.js
index 456f885cca..c138d9ccd3 100644
--- a/client/gutenberg/extensions/markdown/editor.js
+++ b/client/gutenberg/extensions/markdown/editor.js
@@ -12,48 +12,54 @@ import { registerBlockType } from '@wordpress/blocks';
  * Internal dependencies
  */
 import './editor.scss';
+const t = 3000;
 
 Promise.all( [ import( './edit' ), import( './save' ) ] ).then(
-	( [ { default: edit }, { default: save } ] ) =>
-		registerBlockType( 'a8c/markdown', {
-			title: __( 'Markdown' ),
+	( [ { default: edit }, { default: save } ] ) => {
+		console.log( 'Registering in %dms', t );
+		setTimeout( () => {
+			registerBlockType( 'a8c/markdown', {
+				title: __( 'Markdown' ),
 
-			description: (
-				<Fragment>
-					<p>{ __( 'Write your content in plain-text Markdown syntax.' ) }</p>
-					<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
-						{ __( 'Support reference' ) }
-					</ExternalLink>
-				</Fragment>
-			),
+				description: (
+					<Fragment>
+						<p>{ __( 'Write your content in plain-text Markdown syntax.' ) }</p>
+						<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
+							{ __( 'Support reference' ) }
+						</ExternalLink>
+					</Fragment>
+				),
 
-			icon: (
-				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
-					<rect
-						width="198"
-						height="118"
-						x="5"
-						y="5"
-						ry="10"
-						stroke="#000"
-						strokeWidth="10"
-						fill="none"
-					/>
-					<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
-				</svg>
-			),
+				icon: (
+					<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
+						<rect
+							width="198"
+							height="118"
+							x="5"
+							y="5"
+							ry="10"
+							stroke="#000"
+							strokeWidth="10"
+							fill="none"
+						/>
+						<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
+					</svg>
+				),
 
-			category: 'jetpack',
+				category: 'jetpack',
 
-			keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
+				keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
 
-			attributes: {
-				//The Markdown source is saved in the block content comments delimiter
-				source: { type: 'string' },
-			},
+				attributes: {
+					//The Markdown source is saved in the block content comments delimiter
+					source: { type: 'string' },
+				},
 
-			edit,
+				edit,
 
-			save,
-		} )
+				save,
+			} );
+			console.log( 'Registered' );
+		}, t );
+	}
 );
```
</details>

Reply:
thanks for the experiment. we may have to figure out where best to resolve this as the core project may have the same problem already

it makes sense to reload the editor in case we get new block types

Reply:
For `edit`, it seems you can just invert this and do something along the lines of:

```jsx
registerBlockType( 'my-async-edit-block', {
  edit: <AsyncLoad require="./edit" />
} );
```

We just need to make sure the block is registered synchronously.

I'm not sure whether async loading `save` would work or not.

`AsyncLoad` doesn't appear to work with the SDK at this time for at least two reasons:
- Requires a [specific babel transform](https://github.com/Automattic/wp-calypso/blob/b0975a736b49eab842afd9ccecf272ba4554a844/server/bundler/babel/babel-plugin-transform-wpcalypso-async/index.js). With `import()` as [the recommended async solution](https://webpack.js.org/api/module-methods/#import-) ([#](https://webpack.js.org/api/module-methods/#require-ensure)), should we look at removing that transform?
- The `publicPath` issue highlighted here.